Tests/FluentAssertions.Shared.Specs/CollectionEquivalencySpecs.cs

```diff
@@ -17,6 +17,128 @@ namespace FluentAssertions.Specs
     [TestClass]
     public class CollectionEquivalencySpecs
     {
+        #region General
+
+        [TestMethod]
+        public void When_ShouldAllBeEquivalentTo_subject_is_null_it_should_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            IEnumerable<object> subject = null;
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => subject.ShouldAllBeEquivalentTo(Enumerable.Empty<object>());
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act.ShouldThrow<AssertFailedException>().WithMessage(
+                "Expected subject to be*, but found <null>*");
```

Comment:
This is not the "single object" overload test. The report is:

> Expected subject to be a collection, but found \<null\>.
> 
> With configuration:
> - Use declared types and members
> - Compare enums by value
> - Include all non-private properties
> - Include all non-private fields
> - Match member by name (or throw)
> - Be strict about the order of items in byte arrays

Same applies for "single object" overload as tested below:
```c#
[TestMethod]
public void When_ShouldAllBeEquivalentTo_subject_is_null_and_expectation_is_nonenumerable_it_should_throw()
{
    //-----------------------------------------------------------------------------------------------------------
    // Arrange
    //-----------------------------------------------------------------------------------------------------------
    IEnumerable<DateTime> subject = null;

    //-----------------------------------------------------------------------------------------------------------
    // Act
    //-----------------------------------------------------------------------------------------------------------
    Action act = () => subject.ShouldAllBeEquivalentTo(DateTime.Now);

    //-----------------------------------------------------------------------------------------------------------
    // Assert
    //-----------------------------------------------------------------------------------------------------------
    act.ShouldThrow<AssertFailedException>().WithMessage(
        "Expected subject to be*, but found <null>*");
}
```

Expectation report for a failed "single object" `ShouldAllBeEquivalentTo` will contain repeat of single object that was passed in the API. Please check the following example:
```c#
var subject = new int[] { 1, 2, 3 };
subject.ShouldAllBeEquivalentTo(1);
```

> Expected item[1] to be 1, but found 2.
> Expected item[2] to be 1, but found 3.
> 
> With configuration:
> ...

I would appreciate your feedback.